slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/JdbcMapperTest.scala

```diff
@@ -76,6 +76,83 @@ class JdbcMapperTest extends AsyncTest[JdbcTestDB] {
     )
   }
 
+  def testMutatingUpdate = {
+    case class Data(a: Int, b: String)
+
+    class T1(tag: Tag) extends Table[Data](tag, "T_mutating1") {
+      def a = column[Int]("A")
+      def b = column[String]("B")
+      def * = (a, b).mapTo[Data]
+    }
+    val ts1 = TableQuery[T1]
+
+    class T2(tag: Tag) extends Table[Data](tag, "T_mutating2") {
+      def c = column[Int]("C")
+      def d = column[String]("D")
+      def * = (c, d).mapTo[Data]
+    }
+    val ts2 = TableQuery[T2]
+
+    val updateQ11 = ts1.filter(_.a === 11)
+    val updateQ12 = ts1.filter(_.a === 12).map(_.b)
+    val updateQ13 = ts1.filter(_.a === 13).map(t => (t.a, t.b))
+    val updateQ14 = ts1.filter(_.a === 14)
+    val updateQ15 = ts1.filter(_.a === 15).map(_.b)
+    val updateQ16 = ts1.filter(_.a === 16).map(t => (t.b, t.a))
+    val updateQ21 = ts2.filter(_.c === 22)
+    val updateQ22 = ts2.filter(_.c === 25).map(_.d ++ "T2")
+    val updateQ23 = ts2.filter(_.c === 26).map(t => (t.c + 100, t.d ++ "T2"))
+
+    val q1: SqlDBIO[Int] = updateQ11.update3(updateQ21)
+    val q2: SqlDBIO[Int] = updateQ12.update3(updateQ22)
+    val q3: SqlDBIO[Int] = updateQ13.update3(updateQ23)
+//    val q4: SqlDBIO[Int] = updateQ14.update3(d => (d.a + 200, "tuple"))
+    val q5: SqlDBIO[Int] = updateQ15.update3(d => d ++ "T1")
+    val q6: SqlDBIO[Int] = updateQ16.update3[(Rep[String], Rep[Int])]{ case (b, a) => (b ++ "T1", a + 100) }
+    val q7: SqlDBIO[Int] = updateQ14.update(Data(1, "1"))
+    val q8: SqlDBIO[Int] = updateQ15.update3("T0")
+    val q9: SqlDBIO[Int] = updateQ16.update3("T0", 57)
+
+    // Exclude this so that string casting(eg in hsqldb) may not affect the test against constant sql strings
+    ifCapU(scap.stringIsVarchar) {
+      q1.statements.head shouldBe
```

Comment:
I guess I put this in more as a demonstration of what the users could expect the statements to be like than as a test. The queries are being run and tested as well.